seaice_ecdr/tests/integration/test_initial_daily_ecdr_generation.py

```
@@ -169,3 +169,34 @@ def test_can_drop_fields_from_idecdr_netcdf(
 
     ds = xr.open_dataset(output_path)
     assert cdr_conc_fieldname not in ds.variables.keys()
+
+
+def test_seaice_idecdr_has_tyx_data_vars(
+    sample_idecdr_dataset_nh,
+    sample_idecdr_dataset_sh,
+):
+    """Test that idecdr netcdf has (time, y, x) dims for data fields."""
+    expected_tyx_fields = (
+        "conc",
+        "qa_of_cdr_seaice_conc",
+        "raw_bt_seaice_conc",
+        "raw_nt_seaice_conc",
+        "bt_weather_mask",
+        "nt_weather_mask",
+        "invalid_ice_mask",
+        "spatial_interpolation_flag",
+    )
+    for field_name in expected_tyx_fields:
+        nh_data_shape = sample_idecdr_dataset_nh[field_name].shape
+        try:
+            assert len(nh_data_shape) == 3
```

A bare assert here is really all we need - `pytest` will print out an informative message about the error and how the shape differs from what's expected.

If you do want to raise a unique error, it would be easier to avoid the `try-except` block:

```
if not len(nh_data_shape) == 3:
    print('Custom error message)
    raise SystemExit("...")
```

---

Note you can also provide a custom erorr message this way:

```
assert len(nh_data_shape) == 3, "your custom error message if assertion is `False`"
```

---

The try/except was left over from using that to help me catch all the cases.  

Fixed.

Also...good to know about the custom error message!